lib/web/scripts/install_node.go

```diff
@@ -44,19 +55,161 @@ exit 1
 // to install teleport and join a teleport cluster.
 //
 //go:embed node-join/install.sh
-var installNodeBashScript string
+var installNodeBashScriptRaw string
+
+var installNodeBashScript = template.Must(template.New("nodejoin").Parse(installNodeBashScriptRaw))
+
+// InstallNodeScriptOptions contains the options configuring the install-node script.
+type InstallNodeScriptOptions struct {
+	// Required for installation
+	InstallOptions InstallScriptOptions
+
+	// Required for joining
+	Token      string
+	CAPins     []string
+	JoinMethod types.JoinMethod
+
+	// Required for service configuration
+	Labels        types.Labels
+	LabelMatchers types.Labels
+
+	AppServiceEnabled bool
+	AppName           string
+	AppURI            string
+
+	DatabaseServiceEnabled  bool
+	DiscoveryServiceEnabled bool
+	DiscoveryGroup          string
+}
+
+// GetNodeInstallScript generates an agent installation script which will:
+// - install Teleport
+// - configure the Teleport agent joining
+// - configure the Teleport agent services (currently support ssh, app, database, and discovery)
+// - start the agent
+func GetNodeInstallScript(ctx context.Context, opts InstallNodeScriptOptions) (string, error) {
+	// Computing installation-related values
+
+	// By default, it will use `stable/v<majorVersion>`, eg stable/v12
+	repoChannel := ""
+	installPackageUpdater := false
+
+	switch opts.InstallOptions.AutoupdateStyle {
+	case NoAutoupdate:
+		// aaa
+	case PackageManagerAutoupdate:
+		// Note: This is a cloud-specific repo. We could use the new stable/rolling
+		// repo in non-cloud case, but the script has never support enabling autoupdates
+		// in a non-cloud cluster.
+		// We will prefer using the new updater binary for autoupdates in self-hosted setups.
```

Comment:
Should we enforce that we're on cloud for this path to work, or is that a responsibility of whoever's setting the install options?

Reply:
This should be OK, technically there's no issue even if the caller were to try to select this path in non-cloud. In reality we only remove code using this path in the future, so this will should happen.